ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JClient.java

```diff
@@ -49,10 +52,12 @@ public abstract class Web3JClient {
   protected Web3JClient(
       final EventLogger eventLog,
       final TimeProvider timeProvider,
-      final ExecutionClientEventsChannel executionClientEventsPublisher) {
+      final ExecutionClientEventsChannel executionClientEventsPublisher,
+      final Collection<String> nonCriticalMethods) {
     this.eventLog = eventLog;
     this.timeProvider = timeProvider;
     this.executionClientEventsPublisher = executionClientEventsPublisher;
+    this.nonCriticalMethods.addAll(nonCriticalMethods);
```

Comment:
Why not just `this.nonCriticalMethods = nonCriticalMethods` to be consistent

Reply:
Just to ensure it can't be modified externally. So instead of accepting a reference from the outside, I add the values into my own internal collection. Not really critical here but why not. :)